### PR TITLE
Extend globbing section in moving_around.md

### PR DESCRIPTION
**Pull request by mb21**
from https://github.com/nushell/nushell/issues/10749#issuecomment-1768817807:

> Yep, no glob documentation in the book, perhaps you'd like to take a crack at writing it?

### CODE REVIEW

File: book/moving_around.md [hunk 1]
```diff
@@ -24,7 +24,11 @@ In Nushell, we also support double `*` to talk about traversing deeper paths tha
 
 Here, we're looking for any file that ends with ".md", and the two asterisks further say "in any directory starting from here".
 
-In addition to `*`, there is also the `?` pattern which will match a single character. For example, you can match the word "port" by using the pattern `p???`.
+In other shells (like bash), glob expansion happens in the shell and the invoked program (e.g. `ls` in the example above) receives a list of matched files. However, in Nushell the string you enter is passed as is to the command, and some commands (like `ls`, `mv`, `cp` and `rm`) interpret their input string as a glob pattern. For example the [`ls` command's help page](https://www.nushell.sh/commands/docs/ls.html) shows that it takes the parameter: `pattern <string>: the glob pattern to use (optional)`.
```
Comment (mb21) [inline book/moving_around.md:27]: reading https://github.com/nushell/nushell/issues/9558#issuecomment-1684495547, is it actually a string or a dedicated type and `help ls` is wrong?

File: book/moving_around.md [hunk 2]
```diff
@@ -24,7 +24,11 @@ In Nushell, we also support double `*` to talk about traversing deeper paths tha
 
 Here, we're looking for any file that ends with ".md", and the two asterisks further say "in any directory starting from here".
 
-In addition to `*`, there is also the `?` pattern which will match a single character. For example, you can match the word "port" by using the pattern `p???`.
+In other shells (like bash), glob expansion happens in the shell and the invoked program (`ls` in the example above) receives a list of matched files. In Nushell however, the string you enter is passed "as is" to the command, and some commands (like `ls`, `mv`, `cp` and `rm`) interpret their input string as a glob pattern. For example the [`ls` command's help page](https://www.nushell.sh/commands/docs/ls.html) shows that it takes the parameter: `pattern: the glob pattern to use (optional)`.
```
Comment (mb21) [inline book/moving_around.md:27]: reading https://github.com/nushell/nushell/issues/9558#issuecomment-1684495547, is it actually a string, or a dedicated type and `help ls` is wrong?

**Review by amtoine — COMMENTED** [review]
i read that and it looked correct :ok_hand: 

i'd like to have the input of other folks from the core team just to be sure it's correct :relieved:

**Comment by mb21** [comment]
@amtoine sure, it's mostly paraphrased from https://github.com/nushell/nushell/issues/10749#issuecomment-1768817807 and following comments :)

**Comment by fdncred** [comment]
Thanks!

**Pull request merged**